docs/docusaurus/netlify/functions/createJiraTicketInDocsBoard.js

```diff
@@ -14,34 +15,44 @@ const fullDescription = (description, name, email, selectedValue) => {
     return `*Name:* ${formatOptionalValue(name)}\n*Email:* ${formatOptionalValue(email)}\n*Selected feedback type:* ${formattedSelectedValue}\n\n*Description:* ${description}`;
 }
 
+const httpPostRequest = async (url, body) => {
+    return await fetch(url, {
+        method: "POST",
+        headers: {
+            'Content-Type': 'application/json',
+            'Authorization': 'Basic ' + Buffer.from(process.env.JIRA_API_USER + ":" + process.env.JIRA_API_TOKEN).toString('base64')
+        },
+        body: JSON.stringify(body)
+    })
+}
+
 exports.handler = async (req, context) => {
     const { description, name, email, selectedValue } = JSON.parse(req.body);
     try {
-        const response = await fetch(CREATE_JIRA_ISSUE_ENDPOINT_URL, {
-            method: "POST",
-            headers: {
-                'Content-Type': 'application/json',
-                'Authorization': 'Basic ' + Buffer.from(process.env.JIRA_API_USER + ":" + process.env.JIRA_API_TOKEN).toString('base64')
-            },
-            body: JSON.stringify({
-                "fields": {
-                    "project": {
-                        "id": DOCUMENTATION_BOARD_ID
-                    },
-                    "summary": truncateDescription(description),
-                    "description": fullDescription(description, name, email, selectedValue),
-                    "issuetype": {
-                        "id": STORY_ISSUETYPE_ID
-                    },
-                    "labels": [
-                        "feedback_modal"
-                    ],
-                }
-            })
+        const response = await httpPostRequest(JIRA_ISSUE_ENDPOINT_URL, {
+            "fields": {
+                "project": {
+                    "id": DOCUMENTATION_BOARD_ID
+                },
+                "summary": truncateDescription(description),
+                "description": fullDescription(description, name, email, selectedValue),
+                "issuetype": {
+                    "id": STORY_ISSUETYPE_ID
+                },
+                "labels": [
+                    "feedback_modal"
+                ],
+            }
         })
 
         const result = await response.json()
 
+        await httpPostRequest(`${JIRA_ISSUE_ENDPOINT_URL}/${result.id}/transitions`, {
```

Comment:
Let's confirm that the result contains an ID before we make the second request. Otherwise, we should throw an error. I think there could be cases when we can get a response with no ID if we hit some different request code like 400

Reply:
If we get a request code that's not "ok" like 400, we're already throwing that error by returning the `response.status` in the return of the function so I don't think we need some extra logic to do that :thinking: I'm definitely adding the check of having `result.id` to make the second request though, as not to make an invalid request to the API.